File: src/main/scala/com/github/mauricio/postgresql/encoders/CloseMessageEncoder.scala
package com.github.mauricio.postgresql.encoders

import org.jboss.netty.buffer.{ChannelBuffers, ChannelBuffer}
import com.github.mauricio.postgresql.FrontendMessage


/**
 * User: Maurício Linhares
 * Date: 3/3/12
 * Time: 8:41 PM
 */

object CloseMessageEncoder extends Encoder {

  override def encode(message: FrontendMessage): ChannelBuffer = {
    val buffer = ChannelBuffers.buffer(5)
    buffer.writeByte('X')
    buffer.writeInt(0)

    buffer
  }

}

File: postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/encoders/CloseMessageEncoder.kt
package com.github.jasync.sql.db.postgresql.encoders

import com.github.jasync.sql.db.postgresql.messages.frontend.ClientMessage
import io.netty.buffer.ByteBuf
import io.netty.buffer.Unpooled

object CloseMessageEncoder : Encoder {

  override fun encode(message: ClientMessage): ByteBuf {
    val buffer = Unpooled.buffer(5)
    buffer.writeByte('X'.toInt())
    buffer.writeInt(4)
    return buffer
  }

}
